File: js/activitati.js
// Activitățile din planul lunar (altele decât controalele): instruiri, ședințe, birou, concediu etc.
// Se păstrează pe tabletă (meta „activitati”), intră în backup; raportul lunar le adună alături de controale.
import { fmtDate, fmtDateLong, isISO, todayISO, addDays, MONTHS, sarbatoriLegale, zinelucratoare, ucfirst } from './dates.js';
import { activeNereguli, fineDate, isIncheiat, controlStats, uid, byStartDesc, objectives, parseSuma } from './model.js';
import { esc } from './ui.js';

export const TIPURI_ACTIVITATE = [
  { key: 'instruire', label: 'Instruire / pregătire' },
  { key: 'sedinta', label: 'Ședință' },
  { key: 'birou', label: 'Activitate de birou' },
  { key: 'informare', label: 'Informare publică' },
  { key: 'exercitiu', label: 'Exercițiu / aplicație' },
  { key: 'concediu', label: 'Concediu / liber' },
  { key: 'alta', label: 'Altă activitate' },
];
const TIP = Object.fromEntries(TIPURI_ACTIVITATE.map((t) => [t.key, t]));
export const STARI_ACTIVITATE = { planificat: 'Planificată', efectuat: 'Efectuată', anulat: 'Anulată' };

// Activitate nouă: implicit efectuată dacă ziua a trecut, altfel planificată
export function emptyActivitate(data = todayISO(), today = todayISO()) {
  const acum = new Date().toISOString();
  return {
    id: uid(), tip: 'instruire', data, dataSfarsit: '', ora: '', descriere: '',
    stare: data < today ? 'efectuat' : 'planificat', obs: '', objectiveId: '', createdAt: acum, updatedAt: acum,
  };
}

export function normalizeActivitate(a) {
  const out = { ...emptyActivitate(a?.data || todayISO()), ...a };
  if (!TIP[out.tip]) out.tip = 'alta';
  if (!STARI_ACTIVITATE[out.stare]) out.stare = 'planificat';
  if (!isISO(out.dataSfarsit) || out.dataSfarsit <= out.data) out.dataSfarsit = '';
  return out;
}

export const sfarsitActivitate = (a) => a.dataSfarsit || a.data;
export const zileActivitate = (a) => {
  let n = 1;
  for (let d = a.data; d < sfarsitActivitate(a); d = addDays(d, 1)) n++;
  return n;
};
export const tipLabel = (a) => TIP[a.tip]?.label || 'Altă activitate';
// Titlul afișat: „Ședință: analiza lunară”; la „Altă activitate”, doar descrierea
export function titluActivitate(a) {
  const d = String(a.descriere || '').trim();
  if (a.tip === 'alta') return d || 'Altă activitate';
  return d ? `${tipLabel(a)}: ${d}` : tipLabel(a);
}
export function cand(a) {
  const per = a.dataSfarsit ? `${fmtDate(a.data)} – ${fmtDate(a.dataSfarsit)} (${zileActivitate(a)} zile)` : fmtDate(a.data);
  return a.ora ? `${per}, ora ${a.ora}` : per;
}

// Activitățile dintr-o zi (cele pe mai multe zile apar în fiecare zi)
export const activitatiInZi = (list, d) => list.filter((a) => a.data <= d && sfarsitActivitate(a) >= d)
  .sort((x, y) => (x.ora || '99').localeCompare(y.ora || '99') || x.createdAt.localeCompare(y.createdAt));

// Planificate a căror (ultimă) zi a trecut: de confirmat (efectuată / reprogramată / anulată)
export const deConfirmat = (list, today = todayISO()) => list
  .filter((a) => a.stare === 'planificat' && sfarsitActivitate(a) < today)
  .sort((x, y) => x.data.localeCompare(y.data));

// Zilele libere (weekend + sărbători legale) apar implicit în plan, fără introducere manuală:
// efectuate până azi inclusiv, planificate după. null pentru zilele lucrătoare.
// În celula calendarului: numele scurte, ca să încapă întregi (numele complet apare la ziua selectată și în raport)
const SARB_SCURT = {
  'Sfântul Ioan Botezătorul': 'Sf. Ioan', 'a doua zi de Paște': 'Paște (ziua 2)', 'a doua zi de Rusalii': 'Rusalii (ziua 2)',
  'Adormirea Maicii Domnului': 'Sf. Maria', 'Sfântul Andrei': 'Sf. Andrei', 'a doua zi de Crăciun': 'Crăciun (ziua 2)',
};
const scurt = (nume) => nume.split(' / ').map((x) => SARB_SCURT[x] || x).join(' / ');
export function ziLibera(d, today = todayISO()) {
  const motiv = zinelucratoare(d);
  if (!motiv) return null;
  const sarbatoare = sarbatoriLegale(+d.slice(0, 4)).get(d) || '';
  return { d, motiv, sarbatoare, eticheta: sarbatoare ? ucfirst(scurt(sarbatoare)) : 'Liber', stare: d <= today ? 'efectuat' : 'planificat' };
}
export const eticheteLibera = (z) => (z.sarbatoare ? `Sărbătoare legală: ${ucfirst(z.sarbatoare)}` : `Zi liberă (${z.motiv})`);

const suma = (v) => parseSuma(v) ?? 0;
export const lei = (n) => `${n.toLocaleString('ro-RO', { maximumFractionDigits: 2 })} lei`;

// Raportul lunii: controalele începute în lună, amenzile aplicate în lună (după data aplicării),
// activitățile din lună, pe tipuri și pe stări; lista zi cu zi.
export function raportLunar(controls, activitati, an, luna, today = todayISO()) {
  const pre = `${an}-${String(luna + 1).padStart(2, '0')}`;
  const inLuna = (d) => isISO(d) && d.slice(0, 7) === pre;
  const ctl = controls.filter((c) => inLuna(c.dataInceput)).sort((a, b) => -byStartDesc(a, b));
  const constatate = ctl.reduce((s, c) => s + controlStats(c, today).constatate, 0);
  const amenzi = [];
  let faraData = 0;
  for (const c of controls) {
    for (const n of activeNereguli(c)) {
      if (n.status !== 'nok' || !n.amenda?.aplicata) continue;
      const d = fineDate(c, n);
      if (inLuna(d)) amenzi.push({ c, n, data: d, suma: suma(n.amenda.suma) });
      else if (!d && inLuna(c.dataInceput)) faraData++;
    }
  }
  const act = activitati.filter((a) => a.data.slice(0, 7) <= pre && sfarsitActivitate(a).slice(0, 7) >= pre);
  const peTipuri = TIPURI_ACTIVITATE.map((t) => {
    const ef = act.filter((a) => a.tip === t.key && a.stare === 'efectuat');
    return { ...t, n: ef.length, zile: ef.reduce((s, a) => s + zileInLuna(a, pre), 0) };
  }).filter((t) => t.n);
  const zile = new Map();
  const zi = (d) => { if (!zile.has(d)) zile.set(d, { controale: [], activitati: [] }); return zile.get(d); };
  ctl.forEach((c) => zi(c.dataInceput).controale.push(c));
  act.forEach((a) => zi(a.data.slice(0, 7) === pre ? a.data : `${pre}-01`).activitati.push(a));
  const libere = [];
  const zileLuna = new Date(an, luna + 1, 0).getDate();
  for (let k = 1; k <= zileLuna; k++) {
    const z = ziLibera(`${pre}-${String(k).padStart(2, '0')}`, today);
    if (!z) continue;
    // lucrată = în ziua liberă a început un control sau s-a efectuat o activitate (alta decât concediul)
    z.lucrata = ctl.some((c) => c.dataInceput === z.d)
      || act.some((a) => a.stare === 'efectuat' && a.tip !== 'concediu' && a.data <= z.d && sfarsitActivitate(a) >= z.d);
    libere.push(z);
  }
  return {
    an, luna, titlu: `${MONTHS[luna][0].toUpperCase()}${MONTHS[luna].slice(1)} ${an}`,
    controale: ctl, incheiate: ctl.filter(isIncheiat).length, constatate,
    amenzi, sumaAmenzi: amenzi.reduce((s, x) => s + x.suma, 0), amenziFaraData: faraData,
    efectuate: act.filter((a) => a.stare === 'efectuat'), planificate: act.filter((a) => a.stare === 'planificat'),
    anulate: act.filter((a) => a.stare === 'anulat'), peTipuri,
    libere, zileLuna, lucratoare: zileLuna - libere.length,
    zile: [...zile.entries()].sort(([a], [b]) => a.localeCompare(b)),
  };
}
function zileInLuna(a, pre) {
  let n = 0;
  for (let d = a.data; d <= sfarsitActivitate(a); d = addDays(d, 1)) if (d.slice(0, 7) === pre) n++;
  return n;
}

// Raportul, în același stil ca Fișa controlului (tipărire / PDF / partajare)
export function raportMarkup(r, controls, now = new Date()) {
  const obNume = new Map(objectives(controls).map((o) => [o.id, o.denumire]));
  const h = [];
  h.push(`<header class="f-head">
    <div class="f-kicker">Plan lunar · raport de activitate</div>
    <h1>${esc(r.titlu)}</h1>
    <div class="f-meta"><span><b>Generat:</b> ${esc(fmtDateLong(todayISO(now)))}</span></div>
    <div class="f-sum f-sum-6">
      <div><b>${r.controale.length}</b><span>${r.controale.length === 1 ? 'control început' : 'controale începute'} (${r.incheiate} ${r.incheiate === 1 ? 'încheiat' : 'încheiate'})</span></div>
      <div><b>${r.constatate}</b><span>${r.constatate === 1 ? 'neregulă constatată' : 'nereguli constatate'}</span></div>
      <div><b>${r.amenzi.length}</b><span>${r.amenzi.length === 1 ? 'amendă aplicată' : 'amenzi aplicate'}${r.amenzi.length ? ` · ${esc(lei(r.sumaAmenzi))}` : ''}</span></div>
      <div><b>${r.efectuate.length}</b><span>${r.efectuate.length === 1 ? 'activitate efectuată' : 'activități efectuate'}</span></div>
      <div><b>${r.planificate.length}</b><span>${r.planificate.length === 1 ? 'activitate planificată' : 'activități planificate'} (neconfirmate)</span></div>
      <div><b>${r.libere.length}</b><span>${r.libere.length === 1 ? 'zi liberă' : 'zile libere'} (${r.lucratoare} lucrătoare)</span></div>
    </div>
  </header>`);
  if (r.peTipuri.length) {
    h.push(`<section><h2>Activități efectuate, pe tipuri</h2><table class="f-table"><thead><tr><th>Tip</th><th>Activități</th><th>Zile</th></tr></thead><tbody>
      ${r.peTipuri.map((t) => `<tr><td>${esc(t.label)}</td><td>${t.n}</td><td>${t.zile}</td></tr>`).join('')}</tbody></table></section>`);
  }
  if (r.libere.length) {
    const sarb = r.libere.filter((z) => z.sarbatoare);
    const ef = r.libere.filter((z) => z.stare === 'efectuat').length;
    const lucrate = r.libere.filter((z) => z.lucrata);
    h.push(`<section><h2>Zile libere</h2><table class="f-table"><thead><tr><th></th><th>Zile</th></tr></thead><tbody>
      <tr><td>Weekend (fără sărbătorile căzute în weekend)</td><td>${r.libere.length - sarb.length}</td></tr>
      <tr><td>Sărbători legale${sarb.length ? `: ${esc(sarb.map((z) => `${z.sarbatoare} (${fmtDate(z.d)})`).join(', '))}` : ''}</td><td>${sarb.length}</td></tr>
      <tr><td>Efectuate (până azi)</td><td>${ef}</td></tr>
      <tr><td>Planificate</td><td>${r.libere.length - ef}</td></tr>
      <tr><td><b>Zile lucrătoare în lună</b></td><td><b>${r.lucratoare}</b></td></tr></tbody></table>
      ${lucrate.length ? `<p>Zile libere în care s-a lucrat (control început sau activitate efectuată): ${esc(lucrate.map((z) => fmtDate(z.d)).join(', '))}.</p>` : ''}</section>`);
  }
  h.push('<section><h2>Zi cu zi</h2>');
  if (!r.zile.length) h.push('<p>Niciun control și nicio activitate în această lună.</p>');
  for (const [d, x] of r.zile) {
    h.push(`<h3>${esc(fmtDateLong(d))}</h3><ul class="f-list">`);
    for (const c of x.controale) {
      const st = controlStats(c, todayISO(now));
      h.push(`<li><b>Control: ${esc(c.denumire || 'Obiectiv fără denumire')}</b> — ${isIncheiat(c) ? `încheiat ${esc(fmtDate(c.dataIncheiere))}` : 'în desfășurare'}; ${st.constatate} ${st.constatate === 1 ? 'neregulă constatată' : 'nereguli constatate'}${st.fines.length ? `; ${st.fines.length} ${st.fines.length === 1 ? 'amendă' : 'amenzi'}` : ''}</li>`);
    }
    for (const a of x.activitati) {
      const ob = a.objectiveId && obNume.get(a.objectiveId);
      h.push(`<li><b>${esc(titluActivitate(a))}</b> — ${esc(cand(a))}${ob ? ` · ${esc(ob)}` : ''} · <i>${esc(STARI_ACTIVITATE[a.stare])}</i>${a.obs?.trim() ? `<br>${esc(a.obs.trim())}` : ''}</li>`);
    }
    h.push('</ul>');
  }
  h.push('</section>');
  if (r.amenzi.length || r.amenziFaraData) {
    h.push(`<section><h2>Amenzi aplicate în lună</h2>${r.amenzi.length ? `<table class="f-table"><thead><tr><th>Data</th><th>Obiectiv</th><th>Seria și nr.</th><th>Suma</th></tr></thead><tbody>
      ${r.amenzi.sort((a, b) => a.data.localeCompare(b.data)).map((x) => `<tr><td>${esc(fmtDate(x.data))}</td><td>${esc(x.c.denumire || '')}</td><td>${esc(x.n.amenda.serieNr || '—')}</td><td>${x.suma ? esc(lei(x.suma)) : '—'}</td></tr>`).join('')}</tbody></table>` : ''}
      ${r.amenziFaraData ? `<p>+ ${r.amenziFaraData} ${r.amenziFaraData === 1 ? 'amendă' : 'amenzi'} din controale neîncheiate, fără data aplicării (nu sunt numărate mai sus).</p>` : ''}</section>`);
  }
  if (r.anulate.length) h.push(`<section><h2>Activități anulate</h2><ul class="f-list">${r.anulate.map((a) => `<li>${esc(titluActivitate(a))} — ${esc(cand(a))}</li>`).join('')}</ul></section>`);
  return h.join('');
}

export const RAPORT_CSS = `
.fisa-doc .f-list { margin: 0 0 6pt; padding-left: 14pt; }
.fisa-doc .f-list li { margin: 2pt 0; }
.fisa-doc .f-sum.f-sum-6 { grid-template-columns: repeat(3, 1fr) !important; }
`;

export function raportDocument(r, controls, fisaCss) {
  return `<!doctype html><html lang="ro"><head><meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<title>Plan lunar – ${esc(r.titlu)}</title>
<style>@page { size: A4; margin: 14mm; } body { margin: 0; padding: 16px; } ${fisaCss} ${RAPORT_CSS}</style></head>
<body><div class="fisa-doc">${raportMarkup(r, controls)}</div></body></html>`;
}
export const raportFileName = (r) => `Plan-lunar-${r.an}-${String(r.luna + 1).padStart(2, '0')}.html`;

File: ios/AgendaKit/Sources/AgendaKit/Activitati.swift
import Foundation

// Portarea din js/activitati.js: activitățile planului lunar, zilele libere, raportul lunii (datele; HTML-ul în etapa 6).

/// Activitate nouă: implicit efectuată dacă ziua a trecut, altfel planificată
public func emptyActivitate(_ data: String = todayISO(), _ today: String = todayISO()) -> Activitate {
    let acum = isoMs()
    return Activitate(JSObiect([
        ("id", .string(uid())), ("tip", "instruire"), ("data", .string(data)), ("dataSfarsit", ""), ("ora", ""), ("descriere", ""),
        ("stare", .string(data < today ? "efectuat" : "planificat")), ("obs", ""), ("objectiveId", ""),
        ("createdAt", .string(acum)), ("updatedAt", .string(acum)),
    ]))
}

public func normalizeActivitate(_ a: JSObiect) -> Activitate {
    let data = a["data"]?.truthy == true ? a.str("data") : todayISO()
    var out = Activitate(emptyActivitate(data).o.combinat(cu: a))
    if K.tipActivitate(out.tip) == nil { out.tip = "alta" }
    if K.stareActivitate(out.stare) == nil { out.stare = "planificat" }
    if !isISO(out.dataSfarsit) || out.dataSfarsit <= out.data { out.dataSfarsit = "" }
    return out
}

public func sfarsitActivitate(_ a: Activitate) -> String { a.dataSfarsit.isEmpty ? a.data : a.dataSfarsit }

public func zileActivitate(_ a: Activitate) -> Int {
    var n = 1
    var d = a.data
    while d < sfarsitActivitate(a) { d = addDays(d, 1); n += 1 }
    return n
}

public func tipLabel(_ a: Activitate) -> String { K.tipActivitate(a.tip) ?? "Altă activitate" }

/// Titlul afișat: „Ședință: analiza lunară”; la „Altă activitate”, doar descrierea
public func titluActivitate(_ a: Activitate) -> String {
    let d = a.descriere.trimJS
    if a.tip == "alta" { return d.isEmpty ? "Altă activitate" : d }
    return d.isEmpty ? tipLabel(a) : "\(tipLabel(a)): \(d)"
}

public func cand(_ a: Activitate) -> String {
    let per = a.dataSfarsit.isEmpty ? fmtDate(a.data) : "\(fmtDate(a.data)) – \(fmtDate(a.dataSfarsit)) (\(zileActivitate(a)) zile)"
    return a.ora.isEmpty ? per : "\(per), ora \(a.ora)"
}

/// Activitățile dintr-o zi (cele pe mai multe zile apar în fiecare zi)
public func activitatiInZi(_ list: [Activitate], _ d: String) -> [Activitate] {
    list.filter { $0.data <= d && sfarsitActivitate($0) >= d }.sortatStabil { x, y in
        let r = compara(x.ora.isEmpty ? "99" : x.ora, y.ora.isEmpty ? "99" : y.ora)
        return r != 0 ? r : compara(x.createdAt, y.createdAt)
    }
}

/// Planificate a căror (ultimă) zi a trecut: de confirmat (efectuată / reprogramată / anulată)
public func deConfirmat(_ list: [Activitate], _ today: String = todayISO()) -> [Activitate] {
    list.filter { $0.stare == "planificat" && sfarsitActivitate($0) < today }.sortatStabil { compara($0.data, $1.data) }
}

// ───────── Zilele libere (weekend + sărbători legale), calculate la afișare ─────────

private let SARB_SCURT: [String: String] = [
    "Sfântul Ioan Botezătorul": "Sf. Ioan", "a doua zi de Paște": "Paște (ziua 2)", "a doua zi de Rusalii": "Rusalii (ziua 2)",
    "Adormirea Maicii Domnului": "Sf. Maria", "Sfântul Andrei": "Sf. Andrei", "a doua zi de Crăciun": "Crăciun (ziua 2)",
]
private func scurt(_ nume: String) -> String {
    nume.components(separatedBy: " / ").map { SARB_SCURT[$0] ?? $0 }.joined(separator: " / ")
}

public struct ZiLibera: Equatable, Sendable {
    public let d: String, motiv: String, sarbatoare: String, eticheta: String
    /// "efectuat" (până azi inclusiv) | "planificat"
    public let stare: String
    public var lucrata: Bool? = nil

    public var json: JSONValue {
        var o = JSObiect([("d", .string(d)), ("motiv", .string(motiv)), ("sarbatoare", .string(sarbatoare)),
                          ("eticheta", .string(eticheta)), ("stare", .string(stare))])
        if let l = lucrata { o["lucrata"] = .bool(l) }
        return .object(o)
    }
}

public func ziLibera(_ d: String, _ today: String = todayISO()) -> ZiLibera? {
    let motiv = zinelucratoare(d)
    if motiv.isEmpty { return nil }
    let sarbatoare = sarbatoriLegale(Int(d.prefix(4)) ?? 0).get(d) ?? ""
    return ZiLibera(d: d, motiv: motiv, sarbatoare: sarbatoare, eticheta: sarbatoare.isEmpty ? "Liber" : ucfirst(scurt(sarbatoare)),
                    stare: d <= today ? "efectuat" : "planificat")
}

public func eticheteLibera(_ z: ZiLibera) -> String {
    z.sarbatoare.isEmpty ? "Zi liberă (\(z.motiv))" : "Sărbătoare legală: \(ucfirst(z.sarbatoare))"
}

// ───────── Raportul lunii ─────────

public struct AmendaInLuna: Sendable {
    public let c: Control, n: Neregula
    public let data: String
    public let suma: Double
}

public struct TipInRaport: Equatable, Sendable {
    public let key: String, label: String
    public let n: Int, zile: Int
}

public struct ZiInRaport: Sendable {
    public var controale: [Control] = []
    public var activitati: [Activitate] = []
}

public struct RaportLunar: Sendable {
    public let an: Int, luna: Int, titlu: String
    public let controale: [Control]
    public let incheiate: Int, constatate: Int
    public let amenzi: [AmendaInLuna]
    public let sumaAmenzi: Double
    public let amenziFaraData: Int
    public let efectuate: [Activitate], planificate: [Activitate], anulate: [Activitate]
    public let peTipuri: [TipInRaport]
    public let libere: [ZiLibera]
    public let zileLuna: Int, lucratoare: Int
    public let zile: [(String, ZiInRaport)]
}

private func pad2(_ n: Int) -> String { n < 10 ? "0\(n)" : "\(n)" }

/// Raportul lunii (`luna` 0–11, ca în web): controalele începute în lună, amenzile aplicate în lună,
/// activitățile pe tipuri și stări, zilele libere, lista zi cu zi.
public func raportLunar(_ controls: [Control], _ activitati: [Activitate], _ an: Int, _ luna: Int, _ today: String = todayISO()) -> RaportLunar {
    let pre = "\(an)-\(pad2(luna + 1))"
    let inLuna = { (d: String) in isISO(d) && d.prefix(7) == pre }
    let ctl = controls.filter { inLuna($0.dataInceput) }.sortatStabil { -byStartDesc($0, $1) }
    let constatate = ctl.reduce(0) { $0 + controlStats($1, today).constatate }
    var amenzi: [AmendaInLuna] = []
    var faraData = 0
    for c in controls {
        for n in activeNereguli(c) where n.status == "nok" && n.amenda.aplicata {
            let d = fineDate(c, n)
            if inLuna(d) { amenzi.append(AmendaInLuna(c: c, n: n, data: d, suma: parseSuma(n.amenda.suma) ?? 0)) }
            else if d.isEmpty && inLuna(c.dataInceput) { faraData += 1 }
        }
    }
    let act = activitati.filter { $0.data.prefix(7) <= pre && sfarsitActivitate($0).prefix(7) >= pre }
    let peTipuri = K.tipuriActivitate.map { t -> TipInRaport in
        let ef = act.filter { $0.tip == t.key && $0.stare == "efectuat" }
        return TipInRaport(key: t.key, label: t.label, n: ef.count, zile: ef.reduce(0) { $0 + zileInLunaActivitate($1, pre) })
    }.filter { $0.n > 0 }
    var zileOrd: [String] = []
    var zileMap: [String: ZiInRaport] = [:]
    func zi(_ d: String, _ f: (inout ZiInRaport) -> Void) {
        if zileMap[d] == nil { zileOrd.append(d); zileMap[d] = ZiInRaport() }
        f(&zileMap[d]!)
    }
    for c in ctl { zi(c.dataInceput) { $0.controale.append(c) } }
    for a in act { zi(a.data.prefix(7) == pre ? a.data : "\(pre)-01") { $0.activitati.append(a) } }
    var libere: [ZiLibera] = []
    let zileLuna = zileInLuna(an, luna + 1)
    for k in 1...zileLuna {
        guard var z = ziLibera("\(pre)-\(pad2(k))", today) else { continue }
        // lucrată = în ziua liberă a început un control sau s-a efectuat o activitate (alta decât concediul)
        z.lucrata = ctl.contains { $0.dataInceput == z.d }
            || act.contains { $0.stare == "efectuat" && $0.tip != "concediu" && $0.data <= z.d && sfarsitActivitate($0) >= z.d }
        libere.append(z)
    }
    let numeLuna = K.luni[luna]
    return RaportLunar(
        an: an, luna: luna, titlu: "\(ucfirst(numeLuna)) \(an)",
        controale: ctl, incheiate: ctl.filter(isIncheiat).count, constatate: constatate,
        amenzi: amenzi, sumaAmenzi: amenzi.reduce(0) { $0 + $1.suma }, amenziFaraData: faraData,
        efectuate: act.filter { $0.stare == "efectuat" }, planificate: act.filter { $0.stare == "planificat" },
        anulate: act.filter { $0.stare == "anulat" }, peTipuri: peTipuri,
        libere: libere, zileLuna: zileLuna, lucratoare: zileLuna - libere.count,
        zile: zileOrd.sorted().map { ($0, zileMap[$0]!) })
}

private func zileInLunaActivitate(_ a: Activitate, _ pre: String) -> Int {
    var n = 0
    var d = a.data
    while d <= sfarsitActivitate(a) {
        if d.prefix(7) == pre { n += 1 }
        d = addDays(d, 1)
    }
    return n
}
